metadata/src/test/java/org/odk/collect/metadata/PropertyManagerTest.kt

```diff
@@ -0,0 +1,114 @@
+package org.odk.collect.metadata
+
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.odk.collect.permissions.PermissionsProvider
+import org.odk.collect.settings.InMemSettingsProvider
+import org.odk.collect.settings.keys.ProjectKeys
+
+class PropertyManagerTest {
+    private val permissionsProvider = mock<PermissionsProvider>()
+    private val deviceDetailsProvider = TestDeviceDetailsProvider()
+    private val settingsProvider = InMemSettingsProvider()
+
+    private lateinit var propertyManager: PropertyManager
+
+    @Before
+    fun setup() {
+        propertyManager = PropertyManager(permissionsProvider, deviceDetailsProvider, settingsProvider)
+    }
+
+    @Test
+    fun `getSingularProperty should return empty string for undefined properties`() {
+        assertThat(propertyManager.getSingularProperty("blah"), equalTo(""))
+    }
+
+    @Test
+    fun `getSingularProperty should require phone state permission for phone number property only`() {
+        propertyManager.reload()
```

Comment:
I don't think this `reload()` is needed?

Reply:
Right, fixed.